### DISCUSSION TITLE
user.is_anonymous - what does this mean?

### DISCUSSION THREAD

**Discussion by benoj** [saleor/saleor#7631] [Q&A]
Hey There,

Have a quick question to clarify....In the BaseUser there is a property is_anonymous (see definition below)

```
    @property
    def is_anonymous(self):
        """
        Always return False. This is a way of comparing User objects to
        anonymous users.
        """
        return False

```

So if a user checkout out effectively as a guest (i.e. had no account) would this be True?

Thanks,

Ben

**Comment by hylarucoder** [comment]
@benoj 

maybe use `is_authenticated` instead is an option

**Comment by maarcingebala** [comment]
@benoj When a user creates a checkout as a guest (without having an account or logging in), the `user` field in the `Checkout` instance will be `None`. In GraphQL resolvers, where we have access to Django's request object and thus to the current user's instance via `info.context.user` - it will be set to `AnonymousUser` (https://docs.djangoproject.com/en/3.2/ref/contrib/auth/#anonymoususer-object) and `is_anonymous` will be `True`.
